modules/ding_permissions/ding_permissions.module

```diff
@@ -1000,12 +1001,10 @@ function ding_permissions_secure_permissions($role) {
  * Implements hook_preprocess_HOOK().
  */
 function ding_permissions_preprocess_html(&$variables) {
-  global $user;
```

Comment:
The resoning was that if the user has the 'access administration menu' but not the 'access administration pages' the display of admin_menu will have no top-level categories:https://www.drupal.org/node/1337864#comment-5493320

But your right. The proper way would be to update permissions such that all users **without** 'access administration pages' will have the 'access toolbar' and all users **with** the 'access administration pages' should have the 'access administration menu' and **NOT** 'access toolbar' (or else they will see two toolbars).

We could use the preprocess to ensure that no user will see toolbars, but if we're carefull with permissions it shouldn't be an issue.

Let me know if I should update the PR with this solution.